concent_api/sci_testing_common.py

```diff
@@ -0,0 +1,261 @@
+import time
+from typing import Callable
+
+from pathlib import Path
+import requests
+from tempfile import mkdtemp
+
+from eth_utils import to_checksum_address
+from ethereum.utils import privtoaddr
+
+from golem_messages.cryptography import ECCx
+from golem_messages.utils import encode_hex
+from golem_sci import contracts
+from golem_sci import JsonTransactionsStorage
+from golem_sci import new_sci_rpc
+from golem_sci import SmartContractsInterface
+from golem_sci.chains import RINKEBY
+
+
+class ValueNotPositiveError(Exception):
+    pass
+
+
+class NotEnoughFoundsError(Exception):
+    pass
+
+
+class ResponseDonateFaucetError(Exception):
+    pass
+
+
+class SCIBaseTest():
+    @staticmethod
+    def _generate_keys() -> ECCx:
+        return ECCx(None)
+
+    def __init__(self, cluster_type: str) -> None:
+        super().__init__()
+        self.setUp(cluster_type)
+        self.request_for_gntb()
+        self.request_for_deposit()
+
+    def setUp(self, cluster_type: str) -> None:
+        self.provider_keys = self._generate_keys()
+        self.requestor_keys = self._generate_keys()
+        self.provider_empty_account_keys = self._generate_keys()
+        self.requestor_empty_account_keys = self._generate_keys()
+        self.CLUSTER_GNT_DEPOSIT_CONTRACT_ADDRESSES = {
+            'devel': '0xcfB81A6EE3ae6aD4Ac59ddD21fB4589055c13DaD',
+            'staging': '0xA172A4B929Ae9589E3228F723CB99508b8c0709a',
+            'test': '0x694667D7787CFca1892606E81734860a617537B2',
+        }
+        self.geth_rinkeby_address = 'https://rinkeby.golem.network:55555'
+        CONTRACT_ADDRESSES = {
+            contracts.GNT: '0x924442A66cFd812308791872C4B242440c108E19',
```

Comment:
All those addresses should be defined somewhere separate, as constants.

Reply:
I don't know if they should. Golem hardcode them in they code as well (https://github.com/golemfactory/golem/blob/develop/golem/config/environments/testnet.py#L41). Except `contracts.GNTDeposit` they are constant for all testnet environments.

Reply:
Maybe they are. But it's overall a good practice not to hard-code your system for addresses and keys. If they do change, it's a real pain to fish them all out from the code.

BTW, some of them they will depend on the environment (`dev`, `staging`, etc.) so they should be part of those common test settings we're going to prepare in concent-deployment-values.

Reply:
@rwrzesien @cameel OK, so how do you want me to do with this constans now? Should they be in separate file? Separate class? Or somewhere inside Concent settings?

Reply:
I think it will be enough to define them at the top of the file for now (with some grouping/comments), if they will be used somewhere in Concent then probably can be moved to its settings/constants.

Reply:
OK, moved to the top of a file.